### ISSUE TITLE
Tests fail because of missing version: is git required?

### ISSUE THREAD

**Issue by yurivict** [duckdb/duckdb#11727]
### What happens?

Some tests fail like this:
```
-------------------------------------------------------------------------------
Test using a remote optimizer pass in case thats important to someone
-------------------------------------------------------------------------------
/usr/ports/databases/duckdb/work/duckdb-0.10.2/test/extension/test_remote_optimizer.cpp:30
...............................................................................

/usr/ports/databases/duckdb/work/duckdb-0.10.2/test/extension/test_remote_optimizer.cpp:135: FAILED:
  REQUIRE( NO_FAIL((con1.Query("LOAD '" "/usr/ports/databases/duckdb/work/.build" "/test/extension/loadable_extension_optimizer_demo.duckdb_extension'"))) )
with expansion:
  false

[234/3040] (7%): test/extension/load_extension.test                             ================================================================================
Query unexpectedly failed! (test/extension/load_extension.test:24)!
================================================================================
LOAD '/usr/ports/databases/duckdb/work/.build/test/extension/loadable_extension_demo.duckdb_extension';
================================================================================
Invalid Input Error: Extension "/usr/ports/databases/duckdb/work/.build/test/extension/loadable_extension_demo.duckdb_extension" (version v..1-dev, platfrom freebsd_amd64) does not match DuckDB loading it (version , platform freebsd_amd64)

-------------------------------------------------------------------------------
test/extension/load_extension.test
-------------------------------------------------------------------------------
/usr/ports/databases/duckdb/work/duckdb-0.10.2/test/sqlite/test_sqllogictest.cpp:201
...............................................................................

test/extension/load_extension.test:24: FAILED:
```

It prints an empty version string: ```(version , platform freebsd_amd64)```.

We don't use git clones, and git executable isn't available during build  (it is replaced by always failing ```false``` command).

Does it run git, fails, ignores the failure, and leaves the version empty?

This is a wrong behavior. The correct behavior is for configure to fail and to suggest the user that he should pass the version using some cmake variable, for example DUCKDB_VERSION.

Also: some compilation command lines use these bogus arguments: ```-DDUCKDB_VERSION="v..1-dev" -DEXTENSION_VERSION=""```.

### To Reproduce

n/a

### OS:

FreeBSD 14.0

### DuckDB Version:

0.10.2

### DuckDB Client:

C++

### Full Name:

Yuri Victorovich

### Affiliation:

n/a

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

No - Other reason (please specify in the issue body)

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by carlopi** [comment]
This need better documentation, and we are open to suggestions on how to make this clearer to developers.

Current DuckDB behaviour is as follow:

* if you specify OVERRIDE_GIT_DESCRIBE to be a formatted like the output of `git describe --tags --long`, for example for v0.10.2 it would be `v0.10.2-0-g1601d94f94`, that value is used to extract both the version, the dev-version, and the hash to be backed in both DuckDB libraries, binaries and extensions.
* if you specify OVERRIDE_GIT_DESCRIBE only partially, like if you provide only `v0.10.2`, the full `git describe --tags --long`-equivalent is reconstructed asking git the current commit, and that string is used as in part 1
* if you DO NOT specify OVERRIDE_GIT_DESCRIBE, AND git is present in the system, `git describe --tags --long` is computed by GIT and the output is used for further processing
* if you DO NOT specify OVERRIDE_GIT_DESCRIB, AND git is NOT available / functional, the string: `v0.0.1-0-g0123456789` is used and a message is logged in the CMake output informing of this.

I tried:
```
docker run -i --rm -v $(pwd):/duckdb --workdir /duckdb alpine:latest <<< "apk add g++ make cmake ninja && GEN=ninja make && ./build/release/test/unittest test/extension/install_extension.test_slow" 2>&1
```
and stuff seems to work appropriately, with the message logged and `v0.0.1` used as version.

**Comment by yurivict** [comment]
1) The ```v0.0.1-0-g0123456789``` seems like a an arbitrary, bogus value.
2) The ```-DDUCKDB_VERSION="v..1-dev"``` value used in compilation command lines doesn't seem valid, and seems to indicate that the above algorithms fails at some point.
3) Isn't it better for configure to fail when both OVERRIDE_GIT_DESCRIBE and git aren't available?
